app/presenters/search_reference_presenter.rb

```diff
@@ -10,12 +10,18 @@ def to_s
   end
 
   def link
-    "/#{APP_SLUG}/#{referenced_class.tableize}/#{referenced_id}"
+    reference_link = "/#{APP_SLUG}/#{referenced_class.tableize}/#{referenced_id}"
```

Comment:
This will clearly work fine but feels a bit complicated because the logic is around the link rather then around the id generation? An alternative would be another method to return the appropriate id? eg

```ruby
def link
  "/#{APP_SLUG}/#{referenced_class.tableize}/#{composite_id}"
end

def composite_id
  case referenced_class
  when 'Subheading'
    "#{referenced_id}-#{productline_suffix}"
  else
    referenced_id
  end
end
```

FYI - why `case` and no `if` / `else` ? Its because its makes it clear it is special casing according to `referenced_class` and sets up a pattern for extending with other types in the future if necessary.

if thats too long then a ternary would have the same effect but without the future extensibility

```ruby
def composite_id
  referenced_class == 'Subheading' ? "#{referenced_id}-#{productline_suffix}" : referenced_id
end
```

Reply:
I agree, it's more readable.